### PR TITLE
feat(Avatar): add icon prop

### PR DESCRIPTION
**Pull request by mnajdova**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12660)

### CODE REVIEW

File: packages/fluentui/docs/src/examples/components/Avatar/Variations/AvatarExampleIcon.shorthand.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Avatar/Variations/AvatarExampleIcon.shorthand.tsx:1]: Should be this example in `Slots` section?

File: packages/fluentui/react-northstar/src/components/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -98,6 +103,15 @@ const Avatar: React.FC<WithAsProp<AvatarProps>> & FluentComponentStaticProps<Ava
           }),
       })}
       {!image &&
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Avatar/Avatar.tsx:105]: In this case there should be a same condition for `image`, i.e. `!icon && Image.create`.

I suggest: 

```
const imageElement = Image.create()
const iconElement = Icon.create()

const hasGlyph = !!image || !!icon

return (
  <JSX>
    {hasGlyph && (imageElement || iconElement)} 
    {!hasGlyph && Label.create()}
  <JSX>
)
```

File: packages/fluentui/react-northstar/src/components/Avatar/Avatar.tsx [hunk 3]
```diff
@@ -98,6 +103,15 @@ const Avatar: React.FC<WithAsProp<AvatarProps>> & FluentComponentStaticProps<Ava
...
+        Box.create(icon, {
+          defaultProps: () =>
+            getA11Props('icon', {
+              title: name,
+              styles: resolvedStyles.icon,
+            }),
+        })}
+      {!image &&
+        !icon &&
         Label.create(label || {}, {
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Avatar/Avatar.tsx:115]: Not related to your changes, but this is not cool as if I will pass there `null` the slot will be still rendered

File: packages/fluentui/react-northstar/src/themes/teams/components/Avatar/avatarStyles.ts [hunk 4]
```diff
@@ -13,6 +13,16 @@ const sizeToPxValue = {
   largest: 96,
 };
 
+const iconSizeToPxValue = {
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Avatar/avatarStyles.ts:16]:
```suggestion
const iconSizeToPxValue: Record<SizeValue, number> = {
```

File: packages/fluentui/react-northstar/src/components/Avatar/Avatar.tsx [hunk 5]
```diff
@@ -24,6 +25,9 @@ export interface AvatarProps extends UIComponentProps {
    */
   accessibility?: Accessibility<never>;
 
+  /** Avatar can contain icon. */
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/Avatar/Avatar.tsx:28]: Please should we add that icon is only rendered if `image` is not present here or in the docs example?

File: packages/fluentui/docs/src/examples/components/Avatar/Variations/AvatarExampleIcon.shorthand.tsx (hunk 1 above)
Comment (mnajdova) [reply packages/fluentui/docs/src/examples/components/Avatar/Variations/AvatarExampleIcon.shorthand.tsx:1]: Probably, but then we should restructure all examples, as we don't even have slots examples here. I will just leave it now after the image example, and we can restructure the examples in a separate PR.

**Pull request merged**